lib/one_model.ex

```diff
@@ -650,27 +649,51 @@ defmodule OneModel do
   #       the dependency which I tried.
   defp build_fields_list(list, %{include: include}, defaults) when include != [] do
     incl = include -- defaults.assoc_fields
-    select_fields(list, incl, include -- incl)
+    select_fields(list, incl, include -- incl, defaults.assoc_fields)
   end
 
   defp build_fields_list(list, %{exclude: exclude}, defaults) do
-    excl = exclude -- defaults.assoc_fields
-    select_fields(list, defaults.fields -- excl, defaults.assoc_fields -- exclude)
+    select_fields(
+      list,
+      defaults.fields -- exclude,
+      assoc_remove(defaults.assoc_fields, exclude),
+      defaults.assoc_fields
+    )
   end
 
-  # defp build_fields_list(list, _, _) do
-  #   list
-  # end
-
   def do_sf(list, fields, assoc_fields) do
     select_fields(list, fields, assoc_fields)
   end
 
-  defp select_fields(list, fields, assoc_fields) do
+  defp select_fields(list, fields, assoc_fields, default_assoc_fields \\ [])
+
+  defp select_fields([], _, _, _) do
+    []
+  end
+
+  defp select_fields([first | _] = list, fields, assoc_fields, default_assoc_fields) do
+    associations =
+      case first do
+        %{__meta__: _} ->
+          first.__struct__.__schema__(:associations) |> Enum.filter(&(&1 in fields))
+
+        _ ->
+          []
+      end
+
+    assoc =
+      Enum.reduce(default_assoc_fields, assoc_fields, fn value, acc ->
+        if assoc_key(value) in associations && !Enum.find(acc, &(&1 == value)),
```

Comment:
```suggestion
        if assoc_key(value) in associations and not Enum.member?(acc, value),
```
I think this is better suited since you're just checking if the value doesn't exist.